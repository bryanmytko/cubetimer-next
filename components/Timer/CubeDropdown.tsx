import { Dispatch, SyntheticEvent } from "react";

enum TimerActionKind {
  TOGGLE = "TOGGLE",
  READY = "READY",
  REMOVE_TIME = "REMOVE_TIME",
  TICK = "TICK",
  PUZZLE_TYPE = "PUZZLE_TYPE",
}

type TimerAction =
  | {
      type:
        | TimerActionKind.TOGGLE
        | TimerActionKind.READY
        | TimerActionKind.TICK;
    }
  | { type: TimerActionKind.REMOVE_TIME; index: number }
  | { type: TimerActionKind.PUZZLE_TYPE; puzzle: string };

interface CubeDropdownProps {
  dispatch: Dispatch<TimerAction>;
}

const CubeDropdown = (props: CubeDropdownProps) => {
  const { dispatch } = props;
  const onChange = (e: SyntheticEvent & { target: HTMLSelectElement }) => {
    dispatch({ type: TimerActionKind.PUZZLE_TYPE, puzzle: e.target?.value });
  };

  return (
    <div className="relative w-full lg:max-w-sm">
      <select
        defaultValue={"3x3"}
        onChange={onChange}
        className="w-full p-2.5 text-gray-500 bg-white border rounded-md shadow-sm outline-none appearance-none focus:border-indigo-600"
      >
        <option value="2x2">2x2</option>
        <option value="3x3">3x3</option>
        <option value="4x4">4x4</option>
        <option value="5x5">5x5</option>
      </select>
    </div>
  );
};

export default CubeDropdown;
