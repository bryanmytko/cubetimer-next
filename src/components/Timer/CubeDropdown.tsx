import { Dispatch, SyntheticEvent } from "react";
import { TimerActionKind } from "../../reducers";

type TimerAction = {
  type: TimerActionKind.PUZZLE_TYPE;
  puzzle: string
};

interface CubeDropdownProps {
  dispatch: Dispatch<TimerAction>;
  inspectionRunning: boolean;
}

const CubeDropdown = (props: CubeDropdownProps) => {
  const { dispatch, inspectionRunning } = props;
  const onChange = (e: SyntheticEvent & { target: HTMLSelectElement }) => {
    dispatch({ type: TimerActionKind.PUZZLE_TYPE, puzzle: e.target?.value });
  };

  return (
    <>
      <label className="mr-2">Cube Size:</label>
      <select
        defaultValue={"3x3"}
        onChange={onChange}
        className="mt-1 p-1.5 border-solid border-r-4 border-transparent rounded-md focus:outline-gray-300"
        data-testid="select"
        disabled={inspectionRunning}
      >
        <option value="2x2">2x2</option>
        <option value="3x3">3x3</option>
        <option value="4x4">4x4</option>
        <option value="5x5">5x5</option>
      </select>
    </>
  );
};

export default CubeDropdown;
