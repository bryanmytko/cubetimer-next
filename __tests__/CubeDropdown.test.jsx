import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import CubeDropdown from "../components/Timer/CubeDropdown";

it("allows the user to change the cube size", () => {
  const { container } = render(<CubeDropdown />);
  expect(container).toMatchSnapshot();
});
