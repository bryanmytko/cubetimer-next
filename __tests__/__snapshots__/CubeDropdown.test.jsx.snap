// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CubeDropdown> renders the component 1`] = `
<div>
  <label
    class="mr-2"
  >
    Cube Size:
  </label>
  <select
    class="mt-1 p-1.5 border-solid border-r-4 border-transparent rounded-md focus:outline-gray-300"
    data-testid="select"
  >
    <option
      value="2x2"
    >
      2x2
    </option>
    <option
      selected=""
      value="3x3"
    >
      3x3
    </option>
    <option
      value="4x4"
    >
      4x4
    </option>
    <option
      value="5x5"
    >
      5x5
    </option>
  </select>
</div>
`;
